ql/src/java/org/apache/hadoop/hive/ql/exec/MapOperator.java

```diff
@@ -673,7 +674,31 @@ private String toErrorMessage(Writable value, Object row, ObjectInspector inspec
             ctx.getIoCxt().setRecordIdentifier(null);//so we don't accidentally cache the value; shouldn't
             //happen since IO layer either knows how to produce ROW__ID or not - but to be safe
           }
-	  break;
+          break;
+        case PARTITION_SPEC_ID:
```

Comment:
Could we do delegate this to the StorageHandler, or the SerDe?
Again, it could be a follow-up PR

Reply:
Unfortunately we don't have the Table object anywhere around this area as far as I can tell, so I'm not sure how we could inject the logic using the storage handler. Besides, this method is called `populateVirtualColumns` where all the other virtual cols are filled out too, so right now I don't see a better place to put it